overlord/configstate/config/transaction.go

```diff
@@ -159,12 +159,17 @@ func (t *Transaction) Get(snapName, key string, result interface{}) error {
 		return err
 	}
 
-	err = GetFromChange(snapName, subkeys, 0, t.changes[snapName], result)
-	if IsNoOption(err) {
-		err = getFromPristine(snapName, subkeys, 0, t.pristine[snapName], result)
+	// commit changes onto pristine configuration, so that get has a complete view of the config.
+	// this doesn't make them visible in the state until real Commit().
+	config, ok := t.pristine[snapName]
+	if !ok {
+		config = make(map[string]*json.RawMessage)
```

Comment:
can we have some unit tests for this?  also with this change  we lose local coverage of GetFromChange but that is still used elsewhere so it would need to grow its own unit tests here, or we would need to reevaluate if need both getFromPristine and it (not sure what the differences are)

Reply:
Ok, this provoked some more changes and one related bugfix along the way:
-  since GetFromChange is only used by snapctl to get interface attributes now, I moved it to ctlcmd helpers and renamed to. This makes coverage report more clear. Both the "get" for transactions (configstate/..) and "get" for interface attributes (hookstate/ctlcmd/..) should have good coverage now. 
- I've realized that we implicitly require attribute argument with `snapctl :plugOrSlot ....`, but don't enforce this; if omitted, we silently return no data. For now I've fixed this by enforcing attribute name(s) when checking args. There is new test check for that.
- with the above the special-casing for "returning complete document" for interface attributes becomes a dead code (previously it would only be hit by snap configuration code, not by interface-related "get").
- added a couple of extra checks to existing "get" tests.

Reply:
Note, we can (maybe we should) fix `snapctl :plugOrSlot <attribute>` not to require attribute and allow `-d` to get complete document. I can address this in a followup; with currently proposed fix I tried to keep it simple and just maintain existing limitation of the command.